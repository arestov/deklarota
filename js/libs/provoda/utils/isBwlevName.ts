const expr = /^bwlev(($)|(\:.+))/

const isBwlevName = (str: string): boolean => {
  return expr.test(str)
}

export default isBwlevName
