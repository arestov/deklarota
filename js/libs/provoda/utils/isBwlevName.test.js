import {describe, it, expect} from '@jest/globals'

import isBwlevName from './isBwlevName'

describe('isBwlevName', () => {
  it.each([
    ['bwlev'],
    ['bwlev:'],
    ['bwlev:rf'],
    [':rf'],
    [':']

  ])('result should match snapshot for "%s"', input => {
    expect(isBwlevName(input)).toMatchSnapshot()
  })
})
