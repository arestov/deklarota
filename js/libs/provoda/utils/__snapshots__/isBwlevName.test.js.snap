// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isBwlevName result should match snapshot for ":" 1`] = `false`;

exports[`isBwlevName result should match snapshot for ":rf" 1`] = `false`;

exports[`isBwlevName result should match snapshot for "bwlev" 1`] = `true`;

exports[`isBwlevName result should match snapshot for "bwlev:" 1`] = `false`;

exports[`isBwlevName result should match snapshot for "bwlev:rf" 1`] = `true`;
